File: Maths_And_Stats/Geometry/Scalar_Product/scalar_product.cpp
#include<bits/stdc++.h>
using namespace std;

void scalar_prod(int [], int [], int);

int main()
{
	int n, a[100], b[100];
	cout << "Enter size: ";
	cin >> n;
	
	cout << "Enter data of a: ";
	for(int i=0; i<n; i++)
		cin >> a[i];
	
	cout << "Enter data of b: ";
	for(int i=0; i<n; i++)
		cin >> b[i];
		
	scalar_prod(a, b, n);
	
	return 0;
}

void scalar_prod(int x[], int y[], int num)
{
	int prod=0;
	for(int j=0; j<num; j++)
	{
		prod += x[j] * y[j];
	}
	
	cout << "Scalar product: " << prod << endl;
}

File: Maths_And_Stats/Geometry/Scalar_Product/Scalar_Product.py
"""
LANGUAGE: PYTHON
AUTHOR: Diadochokinetic
GITHUB: https://github.com/Diadochokinetic
"""

from operator import mul

def Scalar_Product(x, y):
    return sum(map(mul, x, y))

def float_input(string_input):
    return [float(element) for element in string_input.split()]

if __name__ == '__main__':
    a = float_input(input("Please enter a vector containing n elements, e.g. 1 2 3: "))
    b = float_input(input("Please enter another vector containing n elements, e.g. 4 5 6: "))
    print(f'The scalar product for {a} and {b} is: {Scalar_Product(a, b)}')
